dash-renderer/src/APIController.react.js

```diff
@@ -157,6 +187,7 @@ UnconnectedContainer.propTypes = {
     graphs: PropTypes.object,
     layoutRequest: PropTypes.object,
     layout: PropTypes.object,
+    loadingMap: PropTypes.any,
```

Comment:
The loadingMap replaces the `pendingCallbacks` in that it allows knowing the state of all components in the layout tree and in conjunction with `isLoading` (not used here), it allows to know if the app is ready.